src/sentry/notifications/helpers.py

```diff
@@ -186,29 +189,29 @@ def get_values_by_provider_by_type(
     }
 
 
-def transform_to_notification_settings_by_user(
+def transform_to_notification_settings_by_recipient(
     notification_settings: Iterable["NotificationSetting"],
-    users: Iterable["User"],
+    recipients: Union[Set["User"], Set["Team"]],
```

Comment:
In typing terms is iterable more generic (like could be set or array, for example) and now you're just making it more specific?

Reply:
Yeah `Iterable` is better because it's more generic. I think it'd be simpler if I created a shortcut type like:
```python
RecipientIterable = Union[Iterable["User"], Iterable["Team"]]
```